### PR TITLE
Fixes #8788 - DHCP not detecting pingable addresses on Windows

### PR DESCRIPTION
**Pull request by zjherner**
Creates a Windows specific ping command and outputs to NUL.  Otherwise defaults to a Linux style ping.

### CODE REVIEW

File: modules/dhcp/subnet.rb [hunk 1]
```diff
@@ -277,7 +277,13 @@ def tcp_pingable? ip
 
     def icmp_pingable? ip
       # Always shell to ping, instead of using net-ping
-      system("ping -c 1 -W 1 #{ip} > /dev/null")
+	  if PLATFORM =~ /mingw/
+	    # Windows uses different options for ping and does not have /dev/null
+	    system("ping -n 1 -w 1 #{ip} > NUL")
+	  else
+	    # Default to Linux ping options and send to /dev/null
+	    system("ping -c 1 -W 1 #{ip} > /dev/null")
```
Comment (ohadlevy) [inline modules/dhcp/subnet.rb]: AFAIR this used to work on windows, did the syntax changed based on the OS version? e.g. do we need to have another condition based on the windows version?

**Comment by ohadlevy** [comment]
[test]

File: modules/dhcp/subnet.rb (hunk 1 above)
Comment (zjherner) [inline modules/dhcp/subnet.rb]: I don't think it ever worked on windows.  I just checked my 2003 and 2012
servers and the syntax is the same as 2008.

File: modules/dhcp/subnet.rb [hunk 2]
```diff
@@ -277,7 +277,13 @@ def tcp_pingable? ip
...
+	  end
     rescue
       # We failed to check this address so we should not use it
       true
```
Comment (lzap) [inline modules/dhcp/subnet.rb:289]: Wasn't this working because of this (we return `true` in case of error :)

**Comment by lzap** [comment]
Please fix the indentation. [test]

**Comment by lzap** [comment]
Also squash both commits into one in the proper format please ^^

**Comment by dmitri-d** [comment]
[test]

**Comment by lzap** [comment]
Now I haven't tested that on Windows because that would require me to - ehm - find a Windows box. I am fine to merge, harmless. Anyone want to test on Windows?

**Comment by dmitri-d** [comment]
no windows here either; I'm ok with merging this as is.

**Comment by dmitri-d** [comment]
merged in at a3783af. Thanks, @zjherner!

**Pull request closed without merging**